pkg/app/piped/trigger/deployment.go

```diff
@@ -152,3 +162,34 @@ func buildDeployment(
 
 	return deployment, nil
 }
+
+func (t *Trigger) getMentionedAccounts(d *model.Deployment) ([]string, error) {
+	// Find the application repo from pre-loaded ones.
+	repo, ok := t.gitRepos[d.GitPath.Repo.Id]
+	if !ok {
+		t.logger.Warn("detected some applications binding with a non existent repository", zap.String("repo-id", d.GitPath.Repo.Id))
+		return nil, fmt.Errorf("missing repository")
```

Comment:
Better have an error message easier to understand like:
```suggestion
		return nil, fmt.Errorf("unknown repo %q is set to the deployment", d.GitPath.Repo.Id)
```